SimulationAgent/DeviceConnection/Connect.cs

```diff
@@ -41,23 +42,30 @@ public void Setup(IDeviceConnectionActor context, string deviceId, DeviceModel d
         public async Task RunAsync()
         {
             this.log.Debug("Connecting...", () => new { this.deviceId });
+            var start = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds();
 
             try
             {
                 this.context.Client = this.devices.GetClient(this.context.Device, this.deviceModel.Protocol, this.scriptInterpreter);
 
-                var now = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds();
                 await this.context.Client.ConnectAsync();
                 await this.context.Client.RegisterMethodsForDeviceAsync(this.deviceModel.CloudToDeviceMethods, this.context.DeviceState, this.context.DeviceProperties);
                 await this.context.Client.RegisterDesiredPropertiesUpdateAsync(this.context.DeviceProperties);
 
-                var timeSpent = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds() - now;
-                this.log.Debug("Device connected", () => new { this.deviceId, timeSpent });
+                var timeSpentMsecs = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds() - start;
+                this.log.Debug("Device connected", () => new { timeSpentMsecs, this.deviceId });
                 this.context.HandleEvent(DeviceConnectionActor.ActorEvents.Connected);
             }
+            catch (DeviceAuthFailedException e)
+            {
+                var timeSpentMsecs = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds() - start;
```

Comment:
nit: these can also be refactored to have common code.

Reply:
if we refactor this code out, the log would be generated on a different line of code, and we would lose the ability to see the real line number where the exception is generated